receiver/k8sclusterreceiver/collection/containers.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collection
+
+import (
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/utils"
+)
+
+const (
+	// Keys for container properties.
+	containerKeyStatus       = "container.status"
+	containerKeyStatusReason = "container.status.reason"
+
+	// Values for container properties
+	containerStatusRunning    = "running"
+	containerStatusWaiting    = "waiting"
+	containerStatusTerminated = "terminated"
```

Comment:
Should this be a metric or attributes in the resource/properties?

Reply:
The container status would be part of the properties associated with a `container.id`. That way one could group containers by status while doing some analytics on container metrics.

Reply:
Why? What is the rational to have that way vs a proper metric?